slang.h

```diff
@@ -673,10 +677,68 @@ extern "C"
         */
         virtual SLANG_NO_THROW SlangResult SLANG_MCALL loadFile(
             char const*     path,
-            ISlangBlob**    outBlob) = 0;
+            ISlangBlob** outBlob) = 0;
     };
     #define SLANG_UUID_ISlangFileSystem { 0x003A09FC, 0x3A4D, 0x4BA0, 0xAD, 0x60, 0x1F, 0xD8, 0x63, 0xA9, 0x15, 0xAB }
 
+    /* Type that identifies how a path should be interpreted */
+    typedef unsigned int SlangPathType;
+    enum
+    {
+        SLANG_PATH_TYPE_DIRECTORY,     /**< Path specified specifies a directory. */
+        SLANG_PATH_TYPE_FILE,           /**< Path specified is to a file. */
+    };
+
+    /** An extended file system abstraction.
+    
+    Implementing and using this interface over ISlangFileSystem gives much more control over how paths
+    are managed, as well as how it is determined if two files 'are the same'.
+
+    All paths as input char*, or output as ISlangBlobs are always encoded as UTF-8 strings.
```

Comment:
I'm not sure if we want to state that implementers of these kind of callback interfaces should be expected to return null-terminated blobs just like we do. It should ideally be possible for the user to `mmap()` in a file and wrap that in a blob for us to consume, without worrying about null termination.

Ideally Slang should handle both null-terminated and unterminated input, although we'd need to double-check what constraints the current implementation actually places on input source code buffers.

Reply:
Actually the comment is out of date - and internally it 'does the right thing' irrespective of if they put the terminating 0 or not.